crates/bevy_pbr/src/light.rs

```diff
@@ -268,31 +266,131 @@ fn calculate_cascade_bounds(
+    /// than if you were to use fewer cascades and much larger shadow map textures to achieve the
+    /// same quality level.
+    ///
+    /// In case rendered geometry covers a relatively narrow and static depth relative to camera, it may
+    /// make more sense to use fewer cascades and a higher resolution shadow map texture as perspective aliasing
+    /// is not as much an issue. Be sure to adjust `minimum_distance` and `maximum_distance` appropriately.
+    pub fn num_cascades(mut self, n: usize) -> Self {
+        self.num_cascades = n;
+        self
+    }
+
+    /// Sets the minimum shadow distance, which can help improve the texel resolution of the first cascade.
+    /// Areas nearer to the camera than this will likely receive no shadows.
+    ///
+    /// NOTE: Due to implementation details, this usually does not impact shadow quality as much as
+    /// `first_cascade_far_bound` and `maximum_distance`. At many view frustum field-of-views, the
+    /// texel resolution of the first cascade is dominated by the width / height of the view frustum plane
+    /// at `first_cascade_far_bound` rather than the depth of the frustum from `minimum_distance` to
+    /// `first_cascade_far_bound`.
+    pub fn minimum_distance(mut self, d: f32) -> Self {
+        self.minimum_distance = d;
+        self
+    }
+
+    /// Sets the maximum shadow distance.
+    /// Areas further from the camera than this will likely receive no shadows.
+    pub fn maximum_distance(mut self, d: f32) -> Self {
+        self.maximum_distance = d;
+        self
+    }
+
+    /// Sets the far bound of the first cascade, relative to the view origin.
+    /// In-between cascades will be exponentially spaced relative to the maximum shadow distance.
+    /// NOTE: This is ignored if there is only one cascade, the maximum distance takes precedence.
+    pub fn first_cascade_far_bound(mut self, bound: f32) -> Self {
+        self.first_cascade_far_bound = bound;
+        self
+    }
+
+    /// Sets the overlap proportion between cascades.
+    /// The overlap is used to make the transition from one cascade's shadow map to the next
+    /// less abrupt by blending between both shadow maps.
+    pub fn overlap_proportion(mut self, p: f32) -> Self {
+        self.overlap_proportion = p;
+        self
+    }
+
+    /// Returns the cascade config as specified by this builder.
+    pub fn build(&self) -> CascadeShadowConfig {
         assert!(
-            num_cascades > 0,
-            "num_cascades must be positive, but was {num_cascades}",
+            self.num_cascades > 0,
+            "num_cascades must be positive, but was {}",
+            self.num_cascades
         );
         assert!(
-            (0.0..1.0).contains(&overlap_proportion),
-            "overlap_proportion must be in [0.0, 1.0) but was {overlap_proportion}",
+            (0.0..self.first_cascade_far_bound).contains(&self.minimum_distance),
```

Comment:
This but `cargo fmt`ed:
```suggestion
            self.num_cascades == 1 || (0.0..self.first_cascade_far_bound).contains(&self.minimum_distance),
```

It isn't required when using only 1 cascade as `first_cascade_far_bound` is ignored.